components/encryption/src/manager/mod.rs

```diff
@@ -21,22 +24,29 @@ use crate::{Error, Result};
 
 const KEY_DICT_NAME: &str = "key.dict";
 const FILE_DICT_NAME: &str = "file.dict";
+const FILE_DICT_SIZE: usize = 64;
 
 struct Dicts {
-    file_dict: FileDictionary,
-    key_dict: KeyDictionary,
+    file_dicts: Vec<Mutex<FileDictionary>>,
```

Comment:
Let's not split file.dict into 64 files. We can change the data-structure into the following:
```
file_dict: Mutex<FileDictionary>,
sharded_file_dict: [Mutex<FileDictionary>, FILE_DICT_BUCKETS],
```
here `file_dict` is the source of truth and contain the same data as in "file.dict" file. `sharded_file_dict` is a cached copy of `file_dict` and is sharded. 

`get_file` will only need to access `sharded_file_dict`. the other operations (`new_file`, `delete_file`, `link_file` and `rename_file`) need to follow the steps:
1. update `file_dict` and write file, while taking `file_dict` mutex
2. update `sharded_file_dict`, and return

This will be sufficient reduce mutex contention with `get_file` (e.g. for the the split check and heartbeat flow). The lock contention for writes will be resolve with #8212

Reply:
btw. the `sharded_file_dict` basically match the design of `table_cache` in rocksdb, where file handle is cached and by default have 64 shards. the "file.dict" basically match rocksdb's manifest file, where the file state is kept in the manifest log, and only one thread can write to it at a time.

Reply:
Now use `RwLock` to avoid mutex conflict.

Reply:
Using rwlock is indeed a lot simpler, and I understand the improvement over the existing version is now we only hold read lock while writing to file dictionary. But for the small operation like accessing the file dictionary map, rwlock may be inefficient. We can still change to use 2 mutex, one for in-memory `file_dict`, one for writing to the file. Those operations to update the file would need to take the file lock before taking the in-memory `file_dict` lock.